### ISSUE TITLE
New pre-built binary release listener is not working

### ISSUE THREAD

**Issue by marcopin** [bettar/miele-lxiv#30]
I have tried to make MIELE-LXIV listener port work. But port never opens. I have restarted app and nothing happens. I have restarted my machine and nothing happens. Port remains closed. when 
How can I get M-LXIV dicom listener port work ??

Listener option is activated. The port number is available and all my test has been locally. I have used telnet and other tools, but port is not detected as opened. Is there any log file to see if there is a problem with port opening ??
![config](https://user-images.githubusercontent.com/3383909/49330140-0fc4f900-f561-11e8-818d-dce299c9af14.png)

**Comment by bettar** [comment]
Is your firewall set to off in System Preferences, Security&Privacy, Firewall ?

**Comment by marcopin** [comment]
Yes. It is deactivated.

![image](https://user-images.githubusercontent.com/3383909/49330673-42262480-f568-11e8-8357-26cebaa51b56.png)

**Comment by bettar** [comment]
I just tested the DICOM Listener and it seems to work. Are you sure that you clicked the "Allow" button when prompted for allowing incoming connections ?
![2018-12-01 at 18 50](https://user-images.githubusercontent.com/9733270/49331266-93e7a280-f59a-11e8-9921-8cd5b99be3b4.png)

**Comment by marcopin** [comment]
I'm using Mojave

![image](https://user-images.githubusercontent.com/3383909/49331254-7bfb2900-f570-11e8-9cad-3bd1dd343e41.png)

**Comment by bettar** [comment]
I'm testing in 10.14.1 too. It worked for me. I sent an image to the Listener from the same PC, so it worked even with the Firewall setting "on".
How are you sending images to it ?

**Comment by marcopin** [comment]
I'm not sending images. I tested using dcmtk's echoscu and no response came out.

And when I try to do QUERY it shows this alert message:

![image](https://user-images.githubusercontent.com/3383909/49332602-76a8d900-f586-11e8-9d44-65f1785ebfa0.png)

**Comment by marcopin** [comment]
> I just tested the DICOM Listener and it seems to work. Are you sure that you clicked the "Allow" button when prompted for allowing incoming connections ?
> ![2018-12-01 at 18 50](https://user-images.githubusercontent.com/9733270/49331266-93e7a280-f59a-11e8-9921-8cd5b99be3b4.png)

Let me try

**Comment by marcopin** [comment]
> > I just tested the DICOM Listener and it seems to work. Are you sure that you clicked the "Allow" button when prompted for allowing incoming connections ?
> > ![2018-12-01 at 18 50](https://user-images.githubusercontent.com/9733270/49331266-93e7a280-f59a-11e8-9921-8cd5b99be3b4.png)
> 
> Let me try

I checked, turned firewall on (it was OFF),  

![image](https://user-images.githubusercontent.com/3383909/49332829-b376cf00-f58a-11e8-9192-10c78f33557f.png)

That's not the problem... No response from port yet.

**Comment by marcopin** [comment]
I read this on system log:

Dec  1 17:06:52 Marcos-MBP miele-lxiv[7099]: DEPRECATED USE in libdispatch client: dispatch source activated with no event handler set; set a breakpoint on _dispatch_bug_deprecated to debug.

Could be related to the problem?

**Comment by bettar** [comment]
> I'm not sending images. I tested using dcmtk's echoscu and no response came out.
> 
> And when I try to do QUERY it shows this alert message:
> 
> ![image](https://user-images.githubusercontent.com/3383909/49332602-76a8d900-f586-11e8-9d44-65f1785ebfa0.png)

This is telling me that your listener is "off"

**Comment by marcopin** [comment]
Yes. It is telling that listener is off. But in settings is ON. That's why I open this issue.

**Comment by bettar** [comment]
Can you please test the Listener by using DCMTK `dcmsend` command ? That's what I used and it worked. Also, can you tell me the parameters you are using for `echoscu` so I can try to reproduce your problem ?

**Comment by marcopin** [comment]
I have Horos installed in my machine. I set parameters to dicom listener and activate it, restarted, and it works. Then I quit Horos, open Miele using same parameters (ensuring dicom listener is activated), restarted, but I got no response.

![image](https://user-images.githubusercontent.com/3383909/49339562-65091500-f60a-11e8-80a5-a05188ba2194.png)

**Comment by marcopin** [comment]
When using Horos I get response using telnet. But with Miele I have no response (port is not open even when Dicom listener is activated).

**Comment by marcopin** [comment]
This is the result using echoscu:

![image](https://user-images.githubusercontent.com/3383909/49339693-b9f95b00-f60b-11e8-86ea-f58d0709f595.png)

(Port 11112 is not open even dicom listener is activated)

![image](https://user-images.githubusercontent.com/3383909/49339737-781ce480-f60c-11e8-81a6-dee054f7a3b2.png)

**Comment by bettar** [comment]
This is my result of using `echoscu`
![2018-12-02 at 13 28](https://user-images.githubusercontent.com/9733270/49339746-926bb800-f636-11e8-9f20-972abb273592.png)

Try again by adding the peer AETitle. This is the command I'm using
`echoscu -v --call mini-mojave localhost 11112`

**Comment by marcopin** [comment]
Settings:

![image](https://user-images.githubusercontent.com/3383909/49339823-98996e80-f60d-11e8-9f19-bf1472beeafc.png)

Test with echoscu:

![image](https://user-images.githubusercontent.com/3383909/49339830-afd85c00-f60d-11e8-997b-7ef4bdb5fcbb.png)

**Comment by bettar** [comment]
I see. I really wish I could reproduce the problem. Are you able to use Wireshark to see what's going on with the DICOM association ?

Also, you could try adding `--log-level debug` or even `--log-level trace` to the `echoscu` command.

Another thing you can try is doing a port scan of your PC with `/System/Library/CoreServices/Applications/Network Utility.app` and see who is really using port 11112 or at least have an indication of whether the port is open or not.

**Comment by bettar** [comment]
Just another thought, possibly the problem you reported is related to issue https://github.com/bettar/miele-lxiv/issues/29 that is, maybe Gatekeeper is "blocking" the app.

**Comment by marcopin** [comment]
> I see. I really wish I could reproduce the problem. Are you able to use Wireshark to see what's going on with the DICOM association ?
> 
> Also, you could try adding `--log-level debug` or even `--log-level trace` to the `echoscu` command.
> 
> Another thing you can try is doing a port scan of your PC with `/System/Library/CoreServices/Applications/Network Utility.app` and see who is really using port 11112 or at least have an indication of whether the port is open or not.

Result from echoscu with loglevel:

D: $dcmtk: echoscu v3.6.3 2018-02-05 $
D: 
D: DcmDataDictionary: Loading file: /usr/local/Cellar/dcmtk/3.6.3/share/dcmtk/dicom.dic
D: Request Parameters:
D: ====================== BEGIN A-ASSOCIATE-RQ =====================
D: Our Implementation Class UID:      1.2.276.0.7230010.3.0.3.6.3
D: Our Implementation Version Name:   OFFIS_DCMTK_363
D: Their Implementation Class UID:    
D: Their Implementation Version Name: 
D: Application Context Name:    1.2.840.10008.3.1.1.1
D: Calling Application Name:    ECHOSCU
D: Called Application Name:     MIELE-LXIV
D: Responding Application Name: MIELE-LXIV
D: Our Max PDU Receive Size:    16384
D: Their Max PDU Receive Size:  0
D: Presentation Contexts:
D:   Context ID:        1 (Proposed)
D:     Abstract Syntax: =VerificationSOPClass
D:     Proposed SCP/SCU Role: Default
D:     Proposed Transfer Syntax(es):
D:       =LittleEndianImplicit
D: Requested Extended Negotiation: none
D: Accepted Extended Negotiation:  none
D: Requested User Identity Negotiation: none
D: User Identity Negotiation Response:  none
D: ======================= END A-ASSOCIATE-RQ ======================
I: Requesting Association
T: DUL  FSM Table: State: 1 Event: 0
T: DUL  Event:  A-ASSOCIATE request (local user)
T: DUL  Action: AE 1 Transport Connect
T: DUL  FSM Table: State: 4 Event: 16
T: DUL  Event:  Transport connection closed
T: DUL  Action: AA 4 Indicate AP Abort
F: Association Request Failed: 0006:031b Failed to establish association
F: 0006:0317 Peer aborted Association (or never connected)
F: 0006:031c TCP Initialization Error: Connection refused

**Comment by bettar** [comment]
While I'm looking into this issue, which I suspect being related to the Gatekeeper blocking an untrusted app, please try this command to see if it restores the full functionality to the app:

`$ sudo spctl --add --label "My Miele" /Applications/miele-lxiv.app`

**Comment by zdavatz** [comment]
This should work with v7.1.33. Please reference the commit here.
